### ISSUE TITLE
Paths not relative to configured source

### ISSUE THREAD

**Issue by sarahquigley** [wildlyinaccurate/jekyll-responsive-image#33]
I have a project where I'm using the 'source' configuration option offered by Jekyll to change the directory which Jekyll will read files from when serving / building my project. (See https://jekyllrb.com/docs/configuration/ for details.)

I have found that this plugin does not respect this configuration when locating images by their path, and thus gives an 'Invalid image path' error when it should not. It would great if support for this config option could be added to this plugin.

Thanks so much!

**Comment by wildlyinaccurate** [comment]
Thanks for reporting this! I'm working on it right now.

**Comment by sarahquigley** [comment]
That's great @wildlyinaccurate! Thanks for being so quick to address issues. If only all project maintainers were so ... responsive. 😄

**Comment by wildlyinaccurate** [comment]
> so ... responsive

![](http://i.giphy.com/l2SpVy7i0IEIH8RKo.gif)

I've made a start on the groundwork in #34 but it's proving a bit fiddly so I might not have a fix ready today.

**Comment by sarahquigley** [comment]
No worries! It's great that you've made a start on it - it looks like you're making good progress. Look forward to the fix whenever you've got it ready. Thanks again.

**Comment by wildlyinaccurate** [comment]
Okay, I _think_ I've managed it. I only have one site nowadays which I can use to test this stuff though. It would be a huge help if you could install version `1.0.0.pre3` of the gem and let me know if that works as you'd expect it to? Thanks!

**Comment by sarahquigley** [comment]
@wildlyinaccurate Excellent! I can definitely test on my site - either tonight or tomorrow - I'll report back once I've done so. Will be in touch soon with the results. You're a 🌟 .

**Comment by sarahquigley** [comment]
@wildlyinaccurate So I installed version `1.0.0.pre3` on the project where I'm using the source config option. I encountered an issue which I'll describe.

Here is a sample of an image path that is now being generated by version `1.0.0.pre3` - it is not quite formulated right, with the `src/uploads` of the path repeated twice:
`/Users/sarah/Code/milk-mustache/src/uploads/resized/src/uploads/posts/recipes/2012-05-05-rainbow-salad-with-bulgur-wheat-and-beans/480`

![screen shot 2016-09-26 at 22 17 08](https://cloud.githubusercontent.com/assets/1569881/18858192/295f92cc-8438-11e6-93f8-4da177b20ce4.png)

So some details of my configuration:
My configured source is a directory `src`.
Here is my config for `jekyll-responsive-image`.
```
# https://github.com/wildlyinaccurate/jekyll-responsive-image
responsive_image:
  template: _includes/image.html
  default_quality: 90
  sizes:
    - width: 480
    - width: 620
    - width: 768
    - width: 1000
    - width: 1400
    - width: 2000
  base_path: uploads
  output_path_format: uploads/resized/%{dirname}/%{width}/%{basename}
```
So unless I have misunderstood how to correctly configure this plugin, I think this fix still has a few issues to be ironed out.

Thanks so much again for working on this fix - so appreciated!

**Comment by wildlyinaccurate** [comment]
Thanks for testing it out and for all the useful info! Just so I'm clear on how the `source` config is used: is that where _all_ your Jekyll-related stuff goes? Like, the template lives in `src/_includes/image.html`?

**Comment by wildlyinaccurate** [comment]
Oh, and I'm guessing from your screenshot that you have an image in `src/uploads/posts/recipes/2012-05-05-rainbow~/DSCN0712.jpg`, and you want the resized versions to be in `src/uploads/resized/posts/recipes/2012-05-05-rainbow~/$width/DSCN0712.jpg`?

**Comment by sarahquigley** [comment]
@wildlyinaccurate Yes, that's right - the `source` config specifies the location of all my Jekyll project files - everything Jekyll will use to compile the site is in that directory. And you're also correct in the location which I want the resized versions to be in.

If it helps, this is the branch of my project where I installed `1.0.0.pre3` of this plugin:
https://github.com/finiteadventures/milk-mustache/tree/upgrade-jekyll-responsive-image
Perhaps that will help clarify things.

(I have a bit of a custom set up that I'm trying out on this project - using a Grunt on top of Jekyll's own build system to do a few extra bits and bobs, mostly around dependency injection and unit testing.)

**Comment by wildlyinaccurate** [comment]
Oh that repo is a huge help, thanks a bunch! I'm gonna be on a boat with no internet for 8 days from Friday so I'll try and sort this by tomorrow :)

**Comment by sarahquigley** [comment]
No problem - glad to be able to help! Sounds like a fun trip you have planned.

**Comment by sarahquigley** [comment]
Hope you had a great trip! Any chance of finalising these changes in the next week?

Thanks so much!

**Comment by wildlyinaccurate** [comment]
Hey, I'm sorry I didn't finish this before I went away.

I just published 1.0.0.pre4, and managed to get it working with your project after some small changes (basically just removing `src/` from the config and the image template). I _think_ that makes sense, because you'd expect every path to be relative to the source directory... What do you think?

Here's the full diff of my changes to your project:

```diff
--- a/src/_config.yml
+++ b/src/_config.yml
@@ -76,7 +76,7 @@ owner:

 # https://github.com/wildlyinaccurate/jekyll-responsive-image
 responsive_image:
-  template: src/_includes/image.html
+  template: _includes/image.html
   default_quality: 90
   sizes:
     - width: 480
@@ -85,5 +85,5 @@ responsive_image:
     - width: 1000
     - width: 1400
     - width: 2000
-  base_path: src/uploads
-  output_path_format: src/uploads/resized/%{dirname}/%{width}/%{basename}
+  base_path: uploads
+  output_path_format: uploads/resized/%{dirname}/%{width}/%{basename}
diff --git a/src/_includes/image-group.html b/src/_includes/image-group.html
index 0da0a49..c9780f7 100644
--- a/src/_includes/image-group.html
+++ b/src/_includes/image-group.html
@@ -1,4 +1,4 @@
-{% capture imagePath %}src/{{ site.uploads_path }}/{{ include.entry.layout }}s{% if include.entry.category %}/{{ include.entry.category }}{% endif %}/{% if include.entry.date %}{{ include.entry.date | date: "%Y-%m-%d" }}-{% endif %}{{ include.entry.title | slugify }}/{% endcapture %}
+{% capture imagePath %}{{ site.uploads_path }}/{{ include.entry.layout }}s{% if include.entry.category %}/{{ include.entry.category }}{% endif %}/{% if include.entry.date %}{{ include.entry.date | date: "%Y-%m-%d" }}-{% endif %}{{ include.entry.title | slugify }}/{% endcapture %}
 {% assign images = include.images | split: ' ' %}

 <div class="image-group">
diff --git a/src/_includes/image.html b/src/_includes/image.html
index d849e91..8dd0ee1 100644
--- a/src/_includes/image.html
+++ b/src/_includes/image.html
@@ -6,5 +6,5 @@

 <div class="image-container" style="flex: {{ original.width | plus: 0.0 | divided_by: original.height }} 1 0%">
   {% assign mainImage = resized | where: "width", 2000 | first %}
-  <img src="{{ site.url }}/{{ mainImage.path | remove: 'src/' }}" alt="{{ alt }}" srcset="{{ srcset | strip_newlines | remove: 'src/' }} {{ site.url }}/{{ original.path | remove: 'src/'}} {{ original.width }}w">
+  <img src="{{ site.url }}/{{ mainImage.path }}" alt="{{ alt }}" srcset="{{ srcset | strip_newlines }} {{ site.url }}/{{ original.path }} {{ original.width }}w">
 </div>
```

**Comment by sarahquigley** [comment]
Hey no worries @wildlyinaccurate. I just tested `1.0.0.pre4` on a branch I have set up like the diff above - it's working exactly as I expected. The diff above and what you said about the source directory all makes sense  - all those references to `src/` were some hackery I put in place to get the plugin to work temporarily, and I wanted to eliminate them.

I say ship it! 🚢  :shipit: 

Thanks for all your work on this!
